plugin/externalPluginEntry.go

```diff
@@ -419,7 +422,7 @@ func (e *externalPluginEntry) Stream(ctx context.Context) (io.ReadCloser, error)
 	select {
 	case err := <-headerRdrCh:
 		if err != nil {
-			defer wait()
```

Comment:
Is this safe? I recall previous google searches indicating that you always want to read from stdout/stderr before calling wait, which is why I wrapped it in a defer.

Reply:
I don't believe the docs you pointed to were saying that you should wait before reading from the stdout and stderr readers. I think it was saying that wait would exit when they'd been buffered in. But there's some room for ambiguity.

Reply:
Yeah, the ambiguity is why I wrapped that in a defer. https://blog.kowalczyk.info/article/wOYk/advanced-command-execution-in-go-with-osexec.html also seems to indicate that wait should be called once stdout/stderr were read.

Reply:
Oh, we are using pipes. So looks like we might need to cancel the context but still defer wait to make sure we capture everything. Previously I don't think the 5s timeout worked if the plugin just hung, because it would wait for the pipes to close but they never would.